eng/validation/build-arcadewithrepo.ps1

```diff
@@ -195,52 +210,105 @@ function Git-Command($repoName) {
     }
 }
 
-## If able to retrieve a build, get the SHA that it was built from
-$sha = Get-LastKnownGoodBuildSha -buildDefinitionId $buildDefinitionId
-
-## Clone the repo from git
-GitHub-Clone $targetRepoName
-
-## Create a branch from the repo with the given SHA. 
+## Global Variables
+$githubUri = Get-Github-RepoAuthUri $githubRepoName
+$azdoUri = Get-AzDO-RepoAuthUri $azdoRepoName
+$remoteName = ($azdoOrg + "-" + $azdoRepoName)
 $targetBranch = "dev/" + $githubUser + "/arcade-" + $arcadeSdkVersion
-$githubUri = Get-Github-RepoAuthUri $targetRepoName
-
 $darcBranchName = "refs/heads/" + $targetBranch
-$darcRepoName = "https://github.com/${githubOrg}/${targetRepoName}"
+$darcGitHubRepoName = "https://github.com/${githubOrg}/${githubRepoName}"
+$darcAzDORepoName = "https://${azdoOrg}.visualstudio.com/${azdoProject}/_git/${azdoRepoName}"
 
-# check to see if branch exists and clean it up if it does
-$branchExists = Git-Command $targetRepoName ls-remote --heads $githubUri refs/heads/$targetBranch
+## If able to retrieve a build, get the SHA that it was built from
+$sha = Get-LastKnownGoodBuildSha
+Write-Host "Last Known Good Build SHA: ${sha}"
+
+## Clone the repo from git
+Write-Host "Cloning '${githubRepoName} from GitHub"
+GitHub-Clone $githubRepoName
+ 
+## Check to see if branch exists and clean it up if it does
```

Comment:
Do you need to do cleanup vs. force pushing?

Reply:
I have to check to see if the branch exists, and I figured it was easier to blow the whole branch away and start fresh, than trying to repurpose an existing branch. If I don't, I get these error messages: 

```
Running 'git push dnceng-dotnet-runtime dev/dotnet-bot/arcade-5.0.0-beta.20175.3' from D:\a\1ed3w0ky.thy\runtime
To https://dev.azure.com/dnceng/internal/_git/dotnet-runtime
 ! [rejected]                dev/dotnet-bot/arcade-5.0.0-beta.20175.3 -> dev/dotnet-bot/arcade-5.0.0-beta.20175.3 (fetch first)
error: failed to push some refs to 'https://dotnet-bot:***@dev.azure.com/dnceng/internal/_git/dotnet-runtime'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Git exited with exit code: 1
```

Reply:
In those cases you could force push. either way works really. Feel free to stick with cleanup.